.github/scripts/runner_determinator.py

```diff
@@ -172,85 +217,181 @@ def is_exception_branch(branch: str) -> bool:
     """
-    Used to dynamically opt in jobs to specific runner-type variants.
+    Parse settings, if any, from the rollout state.
+
+    If the issue body contains "---" then the text above that is the settings
+    and the text below is the list of opted in users.
 
-    Returns:
-        The runner-type's variant name if the user has opted in to the feature, otherwise returns an empty string.
-        This variant name is prefixed to the runner-type in the label.
+    If it doesn't contain "---" then the settings are empty and the default values are used.
     """
-    try:
-        userlist = {u.lstrip("#").strip("\n\t@ ") for u in rollout_state.split()}
-        all_opted_in_users = set()
-        for user in userlist:
-            for i in user.split(","):
-                if i == feature:
-                    all_opted_in_users.add(user.split(",")[0])
-        opted_in_requestors = {
-            usr for usr in workflow_requestors if usr in all_opted_in_users
-        }
-
-        if opted_in_requestors:
+    settings_text, _ = extract_settings_user_opt_in_from_text(rollout_state)
+    return parse_settings_from_text(settings_text)
+
+
+def parse_users(rollout_state: str) -> UserOptins:
+    """
+    Parse users from the rollout state.
+
+    """
+    _, users_text = extract_settings_user_opt_in_from_text(rollout_state)
+    return parse_user_opt_in_from_text(users_text)
+
+
+def is_user_opted_in(user: str, user_optins: UserOptins, experiment_name: str) -> bool:
+    """
+    Check if a user is opted into an experiment
+    """
+    return experiment_name in user_optins.get(user, [])
+
+
+def get_runner_prefix(
+    rollout_state: str, workflow_requestors: Iterable[str], is_canary: bool = False
+) -> str:
+    settings = parse_settings(rollout_state)
+    user_optins = parse_users(rollout_state)
+
+    fleet_prefix = ""
+    prefixes = []
+    for experiment_name, experiment_settings in settings.experiments.items():
+        enabled = False
+
+        # Is any workflow_requestor opted in to this experiment?
+        opted_in_users = [
+            requestor
+            for requestor in workflow_requestors
+            if is_user_opted_in(requestor, user_optins, experiment_name)
+        ]
+
+        if opted_in_users:
             log.info(
-                f"Feature {feature} is enabled for {', '.join(opted_in_requestors)}. Using feature {feature}."
+                f"{', '.join(opted_in_users)} have opted into experiment {experiment_name}."
             )
-            return feature
+            enabled = True
         else:
-            log.info(
-                f"Feature {feature} is disabled for {', '.join(workflow_requestors)}. Using fallback \"{fallback}\"."
-            )
-            return fallback
+            # If no user is opted in, then we randomly enable the experiment based on the rollout percentage
+            r = random.randint(1, 100)
+            if r <= experiment_settings.rollout_perc:
```

Comment:
I'm not an expert in the random number generation, but it feels like probability should be potentially continuous, shouldn't it?
```suggestion
            if 100 * random.random() < experiment_settings.rollout_perc:
```

Reply:
Since we're going with floats, yeah.  Just need to make sure we don't randomly generate a 0 and opt someone in at a time that the opt-in rate is supposed to be zero.